src/Tuura/Fantasi/HubRewrite.hs

```diff
@@ -0,0 +1,82 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Tuura.Fantasi.HubRewrite where
+
+import Pangraph
+
+import Data.ByteString.Char8 (pack)
+import Data.ByteString(ByteString)
+import qualified Data.ByteString as BS
+
+import Data.List(sortBy)
+
+import Algebra.Graph hiding (vertexList, edgeList)
+import Algebra.Graph.ToGraph hiding(vertexList, edgeList)
+import qualified Algebra.Graph.ToGraph as Alga
+
+import qualified Data.Set as Set
+
+import Data.Set(toList)
+
+-- Given a hub vertex, and four new vertices, the function replaces the hub with the vertices.
+splitHub :: Ord a => a -> (a, a, a, a) -> Graph a -> Graph a
+splitHub x (x00, x01, x10, x11) g = overlays
+    [ removeVertex x g 
+    , biclique in0 [x00, x01]
+    , biclique in1 [x10, x11]
+    , biclique [x00, x10] out0
+    , biclique [x01, x11] out1 ]
+  where
+    breakInHalf as =    let i = length as `div` 2
+                        in (take i as, drop i as) 
+    (in0 , in1 ) = breakInHalf . toList $ preSet x g
+    (out0, out1) = breakInHalf . toList $ postSet x g
+
+-- Useful alias.
+type Endpoints = (VertexID, VertexID)
```

Comment:
This seems to have been used only once, which I guess doesn't make it very useful. I'd just inline it.